Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/ExposureWindowsFilter.kt

```diff
@@ -0,0 +1,60 @@
+package de.rki.coronawarnapp.risk
+
+import androidx.annotation.VisibleForTesting
+import com.google.android.gms.nearby.exposurenotification.ExposureWindow
+import dagger.Reusable
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.appconfig.ExposureWindowRiskCalculationConfig
+import de.rki.coronawarnapp.risk.result.ExposureWindowDayRisk
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDateUtc
+import kotlinx.coroutines.flow.first
+import org.joda.time.DateTimeZone
+import org.joda.time.Instant
+import javax.inject.Inject
+
+@Reusable
+class ExposureWindowsFilter @Inject constructor(
+    private val appConfigProvider: AppConfigProvider,
+) {
+
+    internal fun filterByAge(
+        config: ExposureWindowRiskCalculationConfig,
+        list: List<ExposureWindow>,
+        nowUtc: Instant
+    ): List<ExposureWindow> = list.filterByAge(
+        maxAgeInDays = config.getMaxEwAgeInDays(),
+        nowUtc = nowUtc
+    )
+
+    internal suspend fun filterDayRisksByAge(
+        list: List<ExposureWindowDayRisk>,
+        calculatedAt: Instant
+    ): List<ExposureWindowDayRisk> {
+        val config = appConfigProvider.currentConfig.first()
+        val deadline = config.getDeadline(calculatedAt).toLocalDateUtc()
+        return list.filter { ewDayRisk ->
+            !ewDayRisk.localDateUtc.isBefore(deadline)
+        }
+    }
+
+    private fun ExposureWindowRiskCalculationConfig.getDeadline(nowUtc: Instant): Instant =
+        nowUtc.minusDays(getMaxEwAgeInDays()).toInstant()
+
+    private fun ExposureWindowRiskCalculationConfig.getMaxEwAgeInDays() =
+        if (maxEncounterAgeInDays > 0) maxEncounterAgeInDays else DEFAULT_EW_AGE_IN_DAYS
```

Comment:
What do you think about having this default value logic in `ExposureWindowRiskCalculationConfigMapper` in case we want to use  `maxEncounterAgeInDays` in other places too?

Reply:
moved the logic to the config mapping as UI should also use the default